src/order.cc

```diff
@@ -3,21 +3,34 @@
 
 namespace tagt::xchg::order {
 
+/**
+ * @return whether this range's lower is actually lower than upper bound
+ */
 bool Range::is_valid() const {
   return this->lower <= this->upper;
 }
 
+/**
+ * @param other pointer to another Range object
+ * @return a Range object with the highest of the .lower and smallest of the .upper
+ */
 Range Range::overlap(const Range& other) const {
   return {
     .lower = std::max(this->lower, other.lower),
     .upper = std::min(this->upper, other.upper)
   };
 }
 
+/**
+ * @return constant Range with the smallest possible price and largest possible price

```

Comment:
Q: Why does it have that behavior?
A: Something to do with how market orders are infinitely elastic (is that the right word?) to price